### DISCUSSION TITLE
Adding Accessory, can we use Switch name?

### DISCUSSION THREAD

**Discussion by andyblac** [mongoose-os-apps/shelly-homekit#1459] [Ideas]
When adding a new accessory can we use the Switch Names from webUI in HomeKit setup (see screenshots), at the moment when we add a new accessory the "Switch Name" is used from the Device ID.

![Screenshot 2020-10-27 at 17 59 55](https://user-images.githubusercontent.com/77936/97342875-b4036780-187e-11eb-9182-5f8c016ab196.jpg)
![IMG_0362](https://user-images.githubusercontent.com/77936/97342886-b665c180-187e-11eb-956b-e39217c4e78a.jpeg)

**Comment by emartife** [comment]
With version 2.7.3 after configuring it you have to reboot, so the correct name appears when it is added to HomeKit

**Comment by andyblac** [comment]
in 2.7.3 the switch name is indeed used, but on adding the accessory it show the wrong switch name, it shows "Device Name", but once added it does show the "Switch Name"

**Comment by rojer** [comment]
hm. i guess we could se accessory name = service name if service is the only one in within the accessory

**Comment by andyblac** [comment]
if adding when in bridge mode, everything works as expected both the hub, accessories are added with all corect names.
![IMG_0514](https://user-images.githubusercontent.com/77936/108605093-e5fa8580-73a9-11eb-9a54-a4147eec77bd.jpeg)
![IMG_0515](https://user-images.githubusercontent.com/77936/108605094-e98e0c80-73a9-11eb-98a0-30a83b67544a.jpeg)
![IMG_0516](https://user-images.githubusercontent.com/77936/108605095-ed219380-73a9-11eb-9101-6343272a3a09.jpeg)
![IMG_0517](https://user-images.githubusercontent.com/77936/108605097-f0b51a80-73a9-11eb-934c-48ace1634f2f.jpeg)
![IMG_0518](https://user-images.githubusercontent.com/77936/108605100-f448a180-73a9-11eb-992b-43ab743c3de5.jpeg)
![IMG_0520](https://user-images.githubusercontent.com/77936/108605186-9072a880-73aa-11eb-881e-f114f5154052.jpeg)
![IMG_0521](https://user-images.githubusercontent.com/77936/108605191-94062f80-73aa-11eb-93c5-50b522f85122.jpeg)

it only when adding a single accessory it shows wrong name:
![IMG_0519](https://user-images.githubusercontent.com/77936/108605112-088c9e80-73aa-11eb-96d5-a3c6b27b6bc3.jpeg)

**Comment by andyblac** [comment]
it could just be a iOS thing?

**Comment by rojer** [comment]
no, it's our thing, will do

**Comment by rojer** [comment]
done, will be in alpha2

**Comment by rojer** [comment]
had to back this one out due to DNS name conflicts, see description in https://github.com/mongoose-os-apps/shelly-homekit/commit/acb57e02caef82438a2c82da5c67265a8c76fda0

**Comment by rojer** [comment]
requires https://github.com/mongoose-os-libs/dns-sd/issues/5

**Comment by andyblac** [comment]
> had to back this one out due to DNS name conflicts, see description in [acb57e0](https://github.com/mongoose-os-apps/shelly-homekit/commit/acb57e02caef82438a2c82da5c67265a8c76fda0)

how about using the original `shelly1-xxxxxx` for A RECORD, but adding an CNAME for the device name ?, i've asked krill to do same in stock firmware as they need to keep `shelly1-xxxxxxx` as A RECORD due the way there app detection works.

https://en.wikipedia.org/wiki/CNAME_record

**Comment by rojer** [comment]
they will still fight for the same name if there is a clash, resulting in confusion.
the spec outlines what the behavior should be, it just needs to be implemented.
